microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/security/ListenerSecurityConfigurationTest.java

```diff
@@ -0,0 +1,238 @@
+import org.jboss.eap.qe.microprofile.openapi.apps.routing.provider.rest.DistrictsResource;
+import org.jboss.eap.qe.microprofile.openapi.apps.routing.provider.services.InMemoryDistrictService;
+import org.jboss.eap.qe.microprofile.openapi.model.OpenApiFilter;
+import org.jboss.eap.qe.microprofile.openapi.model.OpenApiModelReader;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.creaper.ManagementClientProvider;
+import org.jboss.eap.qe.microprofile.tooling.server.log.ModelNodeLogChecker;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.CliException;
+import org.wildfly.extras.creaper.core.online.ModelNodeResult;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.Batch;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+
+/**
+ * Test cases for security scenarios involving HTTP/HTTPS listeners
+ */
+@RunWith(Arquillian.class)
+@ServerSetup({ ListenerSecurityConfigurationTest.OpenApiExtensionSetup.class })
+@RunAsClient
+public class ListenerSecurityConfigurationTest {
+
+    private final static String DEPLOYMENT_NAME = ListenerSecurityConfigurationTest.class.getSimpleName();
+
+    private static String httpsOpenApiUrl;
+    private static OnlineManagementClient onlineManagementClient;
+    private static Path keyStoreFile;
+
+    @Deployment(testable = false)
+    public static WebArchive createServicesProviderDeployment() {
+        WebArchive deployment = ShrinkWrap.create(
+                WebArchive.class,
+                String.format("%s.war", DEPLOYMENT_NAME))
+                .addClasses(ProviderApplication.class)
+                .addClasses(
+                        District.class,
+                        DistrictEntity.class,
+                        DistrictService.class,
+                        InMemoryDistrictService.class,
+                        DistrictsResource.class,
+                        RoutingServiceConstants.class)
+                .addClasses(
+                        OpenApiModelReader.class,
+                        OpenApiFilter.class)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+        return deployment;
+    }
+
+    static class OpenApiExtensionSetup implements ServerSetupTask {
+
+        @Override
+        public void setup(ManagementClient managementClient, String containerId) throws Exception {
+            httpsOpenApiUrl = OpenApiDeploymentUrlProvider.composeHttpsOpenApiUrl();
+            //  MP OpenAPI up
+            onlineManagementClient = ManagementClientProvider.onlineStandalone();
```

Comment:
from online management client you can get https port, in CLI it is `/socket-binding-group=standard-sockets/socket-binding=https/:read-attribute(name=bound-port)

But usually if ports are not standard they are shifted using `-Djboss.socket.binding.port-offset=` 
So from `@ArquillianResource private URL baseURL;` you can easily get https port number - `baseURL.getPort() - 8080 + 8443`

Reply:
Ok, I didn't consider offsets in general, so I think this would lead to a minor refactoring affecting already implemented tests, too.
I'll see how to provide minimal changes in separate commit, thanks for the explanation and hints.